Compiler bug: sometimes extern unions have a tag??

While investigating a mysterious alignment evaluation error with ptrCast, I ran into what might be the root cause:

```zig
pub fn NewThing(comptime T: type) type {
    return struct {
        const Node = extern union {
            next: ?*Node,
            object: T,
        };
        
        pub fn castMe(obj: *T) *Node {
            const aligned_obj = @alignCast(@alignOf(*Node), obj);
            const ref = @ptrCast(*Node, aligned_obj);
            return ref;
        }
    };
}

test "ptrCast align bug" {
    const Thing = NewThing([21]u8);
    const x = Thing.Node(undefined);
    const y = Thing.castMe(&x.object);
}
```

```
ptr_cast_align_bug.zig:19:30: error: accessing union field 'object' while field 'next' is set
    const y = Thing.castMe(&x.object);
                             ^
```

--- a/src/ir.cpp
+++ b/src/ir.cpp
@@ -17952,7 +17952,7 @@ static IrInstruction *ir_analyze_container_field_ptr(IrAnalyze *ira, Buf *field_
                     union_val->special = ConstValSpecialStatic;
                     bigint_init_bigint(&union_val->data.x_union.tag, &field->enum_field->value);
                     union_val->data.x_union.payload = payload_val;
-                } else {
+                } else if (bare_type->data.unionation.layout != ContainerLayoutExtern) {
                     TypeUnionField *actual_field = find_union_field_by_tag(bare_type, &union_val->data.x_union.tag);
                     if (actual_field == nullptr)
                         zig_unreachable();

--- a/test/stage1/behavior/union.zig
+++ b/test/stage1/behavior/union.zig
@@ -511,3 +511,13 @@ test "union with comptime_int tag" {
     };
     comptime expect(@TagType(@TagType(Union)) == comptime_int);
 }
+
+test "extern union doesn't trigger field check at comptime" {
+    const U = extern union {
+        x: u32,
+        y: u8,
+    };
+
+    const x = U{ .x = 0x55AAAA55 };
+    comptime expect(x.y == 0x55);
+}

